pkg/cmd/cli/cmd/login.go

```diff
@@ -178,6 +183,43 @@ func (o LoginOptions) Validate(args []string, serverFlag string) error {
 	return nil
 }
 
+func (o LoginOptions) checkConfigIsWritable(f *osclientcmd.Factory) (bool, error) {
+	kubeconfig := o.PathOptions.GetDefaultFilename()
+
+	file, err := os.OpenFile(kubeconfig, os.O_CREATE|os.O_WRONLY, 0600)
+	if err == nil {
+		file.Close()
+		return true, nil
+	}
+
+	// Whole directory does not exist and we cannot create kubeconfig file.
+	if os.IsNotExist(err) {
+		return true, nil
+	}
+
+	errmsg := kubeconfig
+
+	if os.IsPermission(err) {
+		errmsg += ": The kubeconfig file is not writable. Specify a writeable path with --config or by setting $KUBECONFIG."
```

Comment:
I really don't follow. We have `Complete`, `Validate`, and `Run[Login]`. If this check is internalized into `Validate` (as it should be, right?) we just need to emit an error and it will be treated as any other validation error would. The choice of implementing the error output as a string concatenation instead of `errors.New(fmt.Sprintf(...))` is fairly orthogonal to special-casing this with an `os.Exit`. This is just like any other validation error and I don't really see why any special casing was necessary.